### PR TITLE
solution

### PR DESCRIPTION
**Pull request by tykhan**

### CODE REVIEW

File: src/convertToObject.js [hunk 1]
```diff
@@ -12,7 +12,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitedProp = sourceString.split(';');
```
Comment (irenhh) [inline src/convertToObject.js:15]:
```suggestion
  const splittedProp = sourceString.split(';');
```

File: src/convertToObject.js [hunk 2]
```diff
@@ -12,7 +12,19 @@
...
+
+  const objectProp = {};
+
+  splitedProp.map(item => {
+    const eachProp = item.split(':');
+
+    if (eachProp[0] && eachProp[1]) {
+      objectProp[eachProp[0].trim()] = eachProp[1].trim();
+    }
+  });
```
Comment (irenhh) [inline src/convertToObject.js:25]: map is not for side effects, use forEach

Comment (tykhan) [reply src/convertToObject.js:25]: changed
but not quite understood, what does it mean hear 'for side effects'?

Comment (vkryvytskyy) [reply src/convertToObject.js]: 'side effects' means that for each element in array you do some action. Map is for `changing each element in array`

**Pull request closed without merging**